### DISCUSSION TITLE
where is the function "SSL_write_key"

### DISCUSSION THREAD

**Discussion by boost2020** [gojue/ecapture#565] [Q&A] [answered]
I've searched through all OpenSSL and BoringSSL projects, but I haven't been able to find the SSL_write_key function. So, I'd like to ask, where is it?

**Comment by cfc4n** [comment, marked as answer]
There is no function called `SSL_write_key`. The `SSL_write_key` in `SEC("uprobe/SSL_write_key")` is not crucial; the key point lies in the `AttachToFuncName` function within the eBPF loader. Therefore, you just need to focus on what `AttachToFuncName` is.

https://github.com/gojue/ecapture/blob/cf1d6778c9808a6e61ce780cb4c8bbd6939aca33/user/module/probe_openssl_pcap.go#L140-L146
